src/Cli/dotnet/Commands/Tool/Restore/ToolPackageRestorer.cs

```diff
@@ -129,6 +133,44 @@ public bool PackageHasBeenRestored(
                && _fileSystem.File.Exists(toolCommand.Executable.Value);
     }
 
+    private string CheckForNewerVersion(ToolManifestPackage package, FilePath? configFile)
+    {
+        try
+        {
+            // Use wildcard version range to get the latest version
+            var latestVersionRange = VersionRange.Parse("*");
+            
+            var (latestVersion, _) = _toolPackageDownloader.GetNuGetVersion(
```

Comment:
I'm pretty confident this won't add additional unexpected terminal output, so that's good.